include/swift/ClangImporter/CXXMethodBridging.h

```diff
@@ -96,6 +127,24 @@ struct CXXMethodBridging {
     if (output.empty())
       return output;
 
+    if (classifyNameKind() == NameKind::snake) {
+      for (std::size_t i = 0; i < output.size(); i++) {
+        size_t next = i + 1;
```

Comment:
If the last character is an underscore, `output[next]` isn't checked to exist and is erased. We should check the validity of `next`.
```suggestion
        size_t next = i + 1;
        if (output[next] == output.end())
          return output;
```